x-pack/plugins/rollup/__jest__/integration_tests/job_create.test.js

```diff
@@ -0,0 +1,1298 @@
+        });
+
+        it('should display the histogram fields available', async () => {
+          const { tableCellsValues } = getMetadataFromEuiTable('rollupJobHistogramFieldChooser-table');
+
+          expect(tableCellsValues).toEqual([
+            ['a-numericField'],
+            ['b-numericField'],
+          ]);
+        });
+
+        it('should add histogram field to the field list when clicking on it', () => {
+          let { tableCellsValues } = getMetadataFromEuiTable('rollupJobHistogramFieldList');
+          expect(tableCellsValues).toEqual([['No histogram fields added']]); // make sure the field list is empty
+
+          const { rows } = getMetadataFromEuiTable('rollupJobHistogramFieldChooser-table');
+          rows[0].reactWrapper.simulate('click'); // Select first row
+
+          ({ tableCellsValues } = getMetadataFromEuiTable('rollupJobHistogramFieldList'));
+          const [firstRow] = tableCellsValues;
+          expect(firstRow[0]).toEqual('a-numericField');
+        });
+      });
+    });
+
+    describe('fields list', () => {
+      it('should have an empty field list', async () => {
+        await goToStep(4);
+
+        const { tableCellsValues } = getMetadataFromEuiTable('rollupJobHistogramFieldList');
+        expect(tableCellsValues).toEqual([['No histogram fields added']]);
+      });
+
+      it('should have a delete button on each row to remove an histogram field', async () => {
+        // First let's add a term to the list
+        mockIndexPatternValidityResponse({ numericFields });
+        await goToStepAndOpenFieldChooser();
+        const { rows: fieldChooserRows } = getMetadataFromEuiTable('rollupJobHistogramFieldChooser-table');
+        fieldChooserRows[0].reactWrapper.simulate('click');
+
+        // Make sure rows value has been set
+        let { rows: fieldListRows } = getMetadataFromEuiTable('rollupJobHistogramFieldList');
+        expect(fieldListRows[0].columns[0].value).not.toEqual('No histogram fields added');
+
+        const columnsFirstRow = fieldListRows[0].columns;
+        // The last column is the eui "actions" column
+        const deleteButton = columnsFirstRow[columnsFirstRow.length - 1].reactWrapper.find('button');
+        deleteButton.simulate('click');
+
+        ({ rows: fieldListRows } = getMetadataFromEuiTable('rollupJobHistogramFieldList'));
+        expect(fieldListRows[0].columns[0].value).toEqual('No histogram fields added');
+      });
+    });
+
+    describe('interval', () => {
+      const addHistogramFieldToList = () => {
+        findTestSubject('rollupJobShowFieldChooserButton').simulate('click');
+        const { rows } = getMetadataFromEuiTable('rollupJobHistogramFieldChooser-table');
+        rows[0].reactWrapper.simulate('click');
+      };
+
+      beforeEach(async () => {
+        mockIndexPatternValidityResponse({ numericFields });
+        await goToStep(4);
+        addHistogramFieldToList();
+      });
+
+      describe('--> invalid', () => {
```

Comment:
Could we add a more verbose description? I _think_ we're testing all of the different ways the interval input can enter into an invalid state?